### PR TITLE
[3.10] gh-92311: Let frame_setlineno jump over listcomps

### PR DESCRIPTION
**Pull request by sweeneyde**

### CODE REVIEW

File: Lib/test/test_sys_settrace.py [hunk 1]
```diff
@@ -2094,6 +2094,54 @@ def gen():
...
+        c = 3
+
+    @jump_test(8, 2, [2, 6, 2])
+    def test_jump_backward_over_listcomp_v2(output):
+        flag = False                # 1
+        output.append(2)            # 2
+        if flag:                    # 3
+            return                  # 4
+        x = [i for i in range(10)]  # 5
+        flag = True                 # 6
```
Comment (markshannon) [inline Lib/test/test_sys_settrace.py]:
```suggestion
        flag = 6
```

File: Lib/test/test_sys_settrace.py [hunk 2]
```diff
@@ -2094,6 +2094,54 @@ def gen():
...
+        output.append(6)            # 7
```
Comment (markshannon) [inline Lib/test/test_sys_settrace.py]: Numbers don't match. Could you change the arguments to match the line numbers (then you can drop the comments)

File: Lib/test/test_sys_settrace.py [hunk 3]
```diff
@@ -2094,6 +2094,54 @@ def gen():
...
+        c = 3
+
+    @async_jump_test(8, 2, [2, 6, 2])
+    async def test_jump_backward_over_async_listcomp_v2(output):
+        flag = False
+        output.append(2)
+        if flag:
+            return
+        x = [i async for i in asynciter(range(10))]
+        flag = True
```
Comment (markshannon) [inline Lib/test/test_sys_settrace.py]:
```suggestion
        flag = 6
```

File: Lib/test/test_sys_settrace.py [hunk 4]
```diff
@@ -2094,6 +2094,54 @@ def gen():
...
+        output.append(6)
```
Comment (markshannon) [inline Lib/test/test_sys_settrace.py]: numbers should match line numbers for readability.

**Review by markshannon — CHANGES_REQUESTED** [review]
A few nits, but otherwise looks good.
Thanks for fixing this.

**Comment by sweeneyde** [comment]
I have made the requested changes; please review again

**Comment by markshannon** [comment]
Thanks.

**Pull request merged**